src/m365/entra/commands/user/user-list.ts

```diff
@@ -52,18 +55,32 @@ class EntraUserListCommand extends GraphCommand {
   #initOptions(): void {
     this.options.unshift(
       {
-        option: "--type [type]",
-        autocomplete: ["Member", "Guest"]
+        option: '--type [type]',
+        autocomplete: EntraUserListCommand.allowedTypes
       },
-      { option: '-p, --properties [properties]' }
+      {
+        option: '-p, --properties [properties]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.type && !EntraUserListCommand.allowedTypes.some(t => t.toLowerCase() === args.options.type!.toLowerCase())) {
```

Comment:
Thank you for adding this validation. As addition, could we also add an `#initTypes` function?